File: Phase5-TDD/InvertedIndex.Test/DocTest.cs
using System;
using System.Collections.Generic;
using Xunit;
using InvertedIndex.FileHandler;

namespace InvertedIndex.Test 
{
    public class DocTest : IDisposable
    {
        private const string SAMPLE_NAME = "docName";
        private readonly IList<string> SAMPLE_WORDS = new List<string>() {"word1","word2","word3"};

        private Doc _testDoc;

        public DocTest()
        {
            _testDoc = new Doc(SAMPLE_NAME, SAMPLE_WORDS);
        }

        [Fact]
        public void GetNameTest()
        {
            var name = _testDoc.GetName();
            Assert.Equal(SAMPLE_NAME, name);
        }

        [Fact]
        public void GetWordsTest()
        {
            var words = _testDoc.GetWords();
            Assert.Equal(SAMPLE_WORDS, words);
        }

        [Fact]
        public void EqualsTest()
        {
            Assert.Equal(new Doc(SAMPLE_NAME, SAMPLE_WORDS), _testDoc);
        }

        [Fact]
        public void HashCodeTest()
        {
            Assert.Equal(SAMPLE_NAME.GetHashCode(), _testDoc.GetHashCode());
        }

        public void Dispose()
        {
            GC.Collect();
        }
    }
}

File: Inverted-Index/src/tests/DocTest.java
package tests;

import java.util.Arrays;
import java.util.List;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import file_handler.Doc;
import junit.framework.Assert;

public class DocTest {

    private final String SMAPLE_NAME = "docName";
    private final List<String> SMAPLE_WORDS = Arrays.asList("word1", "word2", "word3");

    private Doc doc1;

    @Before
    public void initiTestDoc() {
        doc1 = new Doc(SMAPLE_NAME, SMAPLE_WORDS);
    }

    @Test
    public void getNameTest() {
        var name = doc1.getName();
        Assert.assertEquals(SMAPLE_NAME, name);
    }

    @Test
    public void getWordsTest() {
        var words = doc1.getWords();
        Assert.assertEquals(SMAPLE_WORDS, words);
    }

    @Test
    public void equalsTest() {
        Assert.assertEquals(new Doc(SMAPLE_NAME, SMAPLE_WORDS), doc1);
    }

    @Test
    public void hashSetTets() {
        Assert.assertEquals(SMAPLE_NAME.hashCode(), doc1.hashCode());
    }

    @Test
    public void toStringTest() {
        Assert.assertEquals(SMAPLE_NAME, doc1.toString());
    }
    @After
    public void dispose() {
        doc1 = null;
    }
}
